source/IntegrationTests/Behaviours/IntegrationEvents/GivenAmountPerChargeResultProducedV1ReceivedTests.cs

```diff
@@ -77,44 +73,339 @@ public async Task When_EnergySupplierActorPeeksMessage_Then_ReceivesCorrectNotif
         await GivenIntegrationEventReceived(amountPerChargeResultProducedEvent);
 
         // Act
-        var peekResult = await WhenActorPeeksMessage(ActorNumber.Create("5799999933318"), ActorRole.EnergySupplier, documentFormat);
+        var peekResultsForEnergySupplier = await WhenActorPeeksAllMessages(ActorNumber.Create("5799999933318"), ActorRole.EnergySupplier, documentFormat);
+        var peekResultsForChargeOwner = await WhenActorPeeksAllMessages(ActorNumber.Create("5799999933444"), ActorRole.GridOperator, documentFormat);
```

Comment:
```suggestion
        var peekResultsForChargeOwner = await WhenActorPeeksAllMessages(ActorNumber.Create("ChargeOwner"), ActorRole.GridOperator, documentFormat);
```
Would it be nice to have variables for chargeOwner, energysupplier ?